File: NoPreLL.cpp
#include <iostream>
#include <fstream>
#include <list>
#include <chrono>
#include <algorithm>

int main() {
    std::ifstream file("numbers.txt");
    if (!file) {
        std::cerr << "Failed to open file for reading." << std::endl;
        return 1;
    }
    std::list<int> numbers;
    auto start = std::chrono::high_resolution_clock::now();
    int num;
    while (file >> num) {
        auto it = numbers.begin();
        while (it != numbers.end() && *it < num) {
            ++it;
        }
        numbers.insert(it, num);
    }
    file.close();
    auto end = std::chrono::high_resolution_clock::now();
    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Total time taken to insert " << numbers.size() << " numbers: " << duration << " ms" << std::endl;

    std::ifstream deleteFile("delete.txt");
    if (!deleteFile) {
        std::cerr << "Failed to open delete file for reading." << std::endl;
        return 1;
    }

    start = std::chrono::high_resolution_clock::now();
    int deleteval;
    while (deleteFile >> deleteval) {
        std::list<int>::iterator pos = std::find(numbers.begin(), numbers.end(), deleteval);
        numbers.erase(pos);
    }
    deleteFile.close();
    end = std::chrono::high_resolution_clock::now();
    duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Deleted, " << numbers.size() << " numbers left. Time: " << duration << " ms" << std::endl;
    return 0;
}

// Results
// Total time taken to insert 100000 numbers: 24874 ms
// Deleted, 0 numbers left. Time: 27117 ms

File: NoPreLL.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.LinkedList;

public class NoPreLL {
    public static void main(String[] args) {
        try {
            // Read numbers from file and insert into linked list
            BufferedReader numbersFile = new BufferedReader(new FileReader("numbers.txt"));
            LinkedList<Integer> numbers = new LinkedList<>();
            long startTime = System.nanoTime();
            String line;
            while ((line = numbersFile.readLine()) != null) {
                int num = Integer.parseInt(line);
                int index = 0;
                for (Integer n : numbers) {
                    if (n >= num) {
                        break;
                    }
                    index++;
                }
                numbers.add(index, num);
            }
            numbersFile.close();
            long endTime = System.nanoTime();
            long duration = (endTime - startTime) / 1000000;
            System.out.println("Total time taken to insert " + numbers.size() + " numbers: " + duration + " ms");

            // Delete elements from linked list by value
            BufferedReader deleteFile = new BufferedReader(new FileReader("delete.txt"));
            startTime = System.nanoTime();
            while ((line = deleteFile.readLine()) != null) {
                int idx = numbers.indexOf(Integer.parseInt(line));
                numbers.remove(idx);
            }
            deleteFile.close();
            endTime = System.nanoTime();
            duration = (endTime - startTime) / 1000000;
            System.out.println("Deleted, " + numbers.size() + " numbers left. Time: " + duration + " ms");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

// Results 100k
// best after jit warm-up
// Total time taken to insert 100000 numbers: 68827 ms
// Deleted, 0 numbers left. Time: 70944 ms
// avg
// Total time taken to insert 100000 numbers: 78942 ms     
// Deleted, 0 numbers left. Time: 86152 ms